b2/raw_api.py

```diff
@@ -440,6 +441,38 @@ def update_bucket(
             **kwargs
         )
 
+    def unprintable_to_question(self, string):
+        """Replace unprintable chars with a question mark."""
+        return re.sub(r'[^\x20-\x7E]', '?', string)
+
+    def check_b2_filename(self, filename):
+        """
+        Raise an appropriate exception with details if the filename is unusable.
+
+        See https://www.backblaze.com/b2/docs/files.html for the rules.
+
+        :param filename: A proposed filename in utf-8.
+        :return: None if the filename is usable."""
+        if len(filename) < 1:
+            raise UnusableFileName("Filename must be at least 1 character.")
+        if len(filename) > 1024:
```

Comment:
The limit is 1024 bytes (of UTF-8), not characters.

Here's a Unicode-in-Python learning opportunity.  :-)

Reply:
Good catch!  I'll fix it.